diff --git a/screens/InboxScreen/screenshots/EntryScreen7.tsx b/screens/InboxScreen/screenshots/EntryScreen7.tsx
index 661a8d2..57a5a6c 100644
--- a/screens/InboxScreen/screenshots/EntryScreen7.tsx
+++ b/screens/InboxScreen/screenshots/EntryScreen7.tsx
@@ -26,16 +26,12 @@ export default function EntryScreen7() {
   const [calleeUser, setCalleeUser] = useState<SendBird.User | null>(null);
 
   useEffect(() => {
-    // getCallee('botUserIds.casey').then((callee) => setCalleeUser(callee));
-    async function getCallee() {
-      const userListQuery = sendbird.createApplicationUserListQuery();
-      userListQuery.userIdsFilter = [botUserIds.casey];
-      userListQuery.limit = 1;
-      const [user] = await userListQuery.next();
-      setCalleeUser(user);
+    async function loadCallee() {
+      const casey = await getCallee(botUserIds.casey);
+      setCalleeUser(casey);
     }
 
-    getCallee();
+    loadCallee();
   }, []);
 
   return (
